backend/assignments/1/src/main/java/org/example/Market.java

```
@@ -0,0 +1,104 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Market {
+    private List<Coin> coinsList;
+    private List<Trader> tradersList;
+
+    private Logging log = new Logging();
+    public Market(List<String[]> coinsData, List<String[]> tradersData){
+        coinsList = new ArrayList<>();
+        tradersList = new ArrayList<>();
+        for(String[] tuple:coinsData){
+            long rank = Long.parseLong(tuple[1]);
+            String name = tuple[2];
+            String symbol = tuple[3];
+            double price = Double.parseDouble(tuple[4]);
+            long circulatingSupply = Long.parseLong(tuple[5]);
+            coinsList.add(new Coin(rank,name,symbol,price,circulatingSupply,circulatingSupply));
+        }
+        for(String[] tuple:tradersData){
+            String firstName = tuple[1];
+            String lastName = tuple[2];
+            String phone = tuple[3];
+            String walletAddress = tuple[4];
+            tradersList.add(new Trader(firstName,lastName,phone,walletAddress,0,0));
+        }
+    }
+    public Coin getCoinByName(String coinName){
+        for(Coin coin:coinsList){
+            if(coinName.equalsIgnoreCase(coin.getSymbol())){
+                return coin;
+            }
+        }
+        return null;
```

Lets not return `null` here. Lets throw an exception and catch it.